### ISSUE TITLE
Normal Maps are not working (edit: they work.. sometimes)  

### ISSUE THREAD

**Issue by vurt72** [kirides/GD3D11#164]
i am running The Chronicles of Myrtana. 

I have placed my normal maps (.dds files, no alpha) in

C:\Program Files (x86)\Steam\steamapps\common\TheChroniclesOfMyrtana\system\GD3D11\Textures\Replacements\Normalmaps_TheChroniclesOfMyrtana
i have also tried
C:\Program Files (x86)\Steam\steamapps\common\TheChroniclesOfMyrtana\system\GD3D11\Textures\Replacements\Normalmaps_ChroniclesOfMyrtana

I have no idea what it expects, where it finds this "mod name" which is required, and which of them is the correct one. It might even be the polish name.

If i turn on the normal maps in the renderer i can very vaguely see some black specks, if the lighting is allowing it. I can also see that characters gets a normal map. so its likely just a default normal map texture that applies to everything and not my normal map textures.

And yes i have my own texture pack for CoM that i created and now i created normals as well.

**Comment by vurt72** [comment]
I am guessing that these packs must somehow be "approved" by the engine? 

Because I know for my old texture pack it had to be named replacements_vurt. i just found the file on the internet so i checked how it was made (named).  

Would be great if things like this could be explained on the page. It's really cool that you've made this possible, it looks great, but not cool to not explain almost anything on the page itself.

**Comment by kirides** [comment]
This should still hold true, maybe they broke with a recent version?

https://forum.worldofplayers.de/forum/threads/1546222-Yet-Another-D3D11-Renderer

>  Normalmaps gehören in ein Mod-Spezifisches Verzeichnis in dem Verzeichnis
> "Gothic\system\GD3D11\textures\replacements\"
> 
> Normalmaps für das normale Spiel, kommen in das Verzeichnis "Normalmaps_Original" (ggf. anlegen wenn nicht vorhanden)
> 
> Bei Modifikationen ist lautet der Name "Normalmaps_{MOD}".
> Für Atariar findet man den Namen so heraus:
> 
>     In den "Gothic\System"-Ordner gehen
>     nach einer Datei mit dem Namen "Atariar" ausschau halten.
>     Die Datei die wir suchen heißt "Gothic2_Atariar.ini"
>     Von der Datei entfernen wir nun das ".ini" und es bleibt "Gothic2_Atariar".
>     Unser Verzeichnis lautet damit "Normalmaps_Gothic2_Atariar"
> 
> 
> 
> Diese Art der Normalmaps-Ablage ermöglicht Mod-Spezifische Normalmaps. Z.B. L'Hiver Normalmaps wenn man L'Hiver spielt und keine wenn man normales Gothic spielt.

**Comment by vurt72** [comment]
> This should still hold true, maybe they broke with a recent version?

yeah it's broken. I also let another person try it (he's a programmer + he's just better technically with G2/modding than i am). He says it just doesn't work for this recent version and that he's had it working before.

**Comment by vurt72** [comment]
I made the distortion.dds maps blank to test my theory, and yeah, they're 100% the only normals that affects the textures in any shape, other normal maps does nothing. 

it's easy to get tricked thinking you have it working because of the rather strong effect of those default normals that affects all maps if it doesn't find a .dds for that texture. 

I even tried an earlier version of the renderer and its not working there either. 
path: system\GD3D11\Textures\Replacements\Normalmaps_Original
tried some mountain and ground and leaf textures.

did this ever work really? or was everyone just tricked by the default normal map it applies?

**Comment by kirides** [comment]
i'm glad you could figure this out.

i'll add a note for that sometime in the future (or maybe _someone_ wants to open a PR to mention Normalmaps and their limitations in the readme? :) )

**Comment by vurt72** [comment]
BC3/DXT5 also works (has Alpha).

I have seen many texture packs that uses the diffuse texture in the alpha channel. Very unsure if this really adds anything? I have mine without and it looks fine.

**Comment by vurt72** [comment]
Ok so they've stopped working, and i have no idea why. i did mess around with Union, trying to get that to work when it suddenly didn't work any longer, after that it seems like the normals just decided to not work. 

So I made a reinstall of the game and the latest nightly build of the renderer. Didn't install Union or anything else, just the normal maps and my texture mod. But no, no change any longer with normals on or off. 

My other friend who's also testing this is experiencing the same, it's super easy to break the normals from functioning at all, and impossible to know what even broke it or how to get it to work again.

**Comment by vurt72** [comment]
I have my texture pack with normal maps here if you want to test it out in Chronicles of Myrtana. 

Can you get the normal maps to work?

https://www.nexusmods.com/gothic2/mods/91

**Comment by kirides** [comment]
i'll try to find some time looking at it.

maybe/hopefully it's something simple to fix.

**Comment by kirides** [comment]
@vurt72 can you tell me what is not working? i tried the normalmaps and atleast _something_ is happening.

https://github.com/user-attachments/assets/6d99872f-0494-4b4a-b931-fb6c97e2ae41
https://github.com/user-attachments/assets/d6a43231-951b-4c34-8a10-3493520ce6d4

some normal maps were broken though, as they used RGBA32 textures instead of BC1 or BGRA

you can convert all normal maps using something like this
```
> mkdir converted
> texconv.exe -o converted -f B8G8R8A8_UNORM *.dds
```

for compression use BC1 (or appropriate)
```
> mkdir converted
> texconv.exe -o converted -f BC1_UNORM *.dds
```

**Comment by vurt72** [comment]
Yeah i know, i forgot to resave some of the later ones i did in Photoshop. I have an AI that makes them + saves in .dds, but in the wrong format. 

For some reason it works again now.. i don't know, i reinstalled the entire thing. Maybe it was just me all along that messed up something, they always worked in Gothic 2.

Great to have that in the log though! Much appreciated!
